spinn_front_end_common/utilities/emergency_recovery.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2017-2020 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+
+from spinn_utilities.log import FormatAdapter
+from spinnman.model import ExecutableTargets, CPUInfos
+from spinnman.model.enums import CPUState
+from .iobuf_extractor import IOBufExtractor
+from .globals_variables import get_simulator
+
+logger = FormatAdapter(logging.getLogger(__name__))
+
+
+def _emergency_state_check(txrx, app_id):
+    """
+    :param ~.Transceiver txrx:
+    :param int app_id:
+    """
+    # pylint: disable=broad-except
+    try:
+        rte_count = txrx.get_core_state_count(
+            app_id, CPUState.RUN_TIME_EXCEPTION)
+        watchdog_count = txrx.get_core_state_count(app_id, CPUState.WATCHDOG)
+        if rte_count or watchdog_count:
+            logger.warning(
+                "unexpected core states (rte={}, wdog={})",
+                txrx.get_cores_in_state(None, CPUState.RUN_TIME_EXCEPTION),
+                txrx.get_cores_in_state(None, CPUState.WATCHDOG))
+    except Exception:
```

Comment:
too broad exception

Reply:
This is just a copy of the source, nothing more.  Any exception is absolutely fine here though.

Reply:
bah, good pratice to list them. ;-)